### PR TITLE
Change/9701 s2s reporting user

### LINKED ISSUE #9701: Cleanup reporting user validation
### Bug Description
We discovered  in #9696 that reporting user validation is probably inconsistent and is only triggered for Contacts.
Citing @leventegal-she 
> Please create a ticket to cleanup the reporting user validation. It's strange that only contact validation fails if it has no reporting user.

Behavior should be consistent for all entities.

~Please also adjust the tests (e.g., `testShareContact()`) accordingly~ Test works

---

Found:
- `EventFacade`
- `EventParticipateFacade`
- `ImmunizationFacade`
- `CaseFacadeEjb`

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #9701
The PR does the following:

- Do not set `reportingUser` to null where it is required while building the S2S share
- Add generic method to build the DTO while sharing

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/entities/event/EventShareDataBuilder.java [hunk 1]
```diff
@@ -67,20 +64,18 @@ public void doBusinessValidation(SormasToSormasEventDto sormasToSormasEventDto)
...
+		Pseudonymizer pseudonymizer = dataBuilderHelper.createPseudonymizer(requestInfo);
 		return getEventPreview(event, pseudonymizer);
 	}
 
-	private EventDto getEventDto(Event event, Pseudonymizer pseudonymizer) {
-		EventDto eventDto = eventFacade.convertToDto(event, pseudonymizer);
+	@Override
+	protected EventDto getDto(Event event, Pseudonymizer pseudonymizer) {
 
-		eventDto.setReportingUser(null);
```
Comment (BarnaBartha) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/entities/event/EventShareDataBuilder.java:79]: it is a bit confusing what the overall change wants to be
looking at other parts i would say we need this nulling of the reporting user
some minimal unit tests would shed light on the whole thing as the ticket you are referencing is also a bit lacking in description

Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/entities/event/EventShareDataBuilder.java:79]: Hey, the nulling of the user is exactly what causes the transmission to fail. https://github.com/hzi-braunschweig/SORMAS-Project/pull/9696 already did this for contacts. I can create a unit test.

Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/entities/event/EventShareDataBuilder.java:79]: tests added

**Comment by JonasCir** [comment]
I added test for cases, contacts, events. Event test also includes event participant and samples, cases also covers immunizations.

**Review by BarnaBartha — APPROVED** [review]

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**